### DISCUSSION TITLE
Dominant, always topmost WinForms AxHost element in Avalonia NativeControlHost

### DISCUSSION THREAD

**Discussion by StayMeaty** [AvaloniaUI/Avalonia#12643] [Q&A] [answered]
I am contacting you today with a very specific problem. I am using Microsoft's MstscAx interface to establish a remote desktop connection. This interface provides me with a WinForms AxHost element, which I in turn place using an Avalonia.Controls.NativeControlHost in an Avalonia window. So far, everything works as it should. My problem is that this AxHost element is absolutely topmost and I just can't find a way to make it background. It completely ignores the order in the VirtualTree and all controls that are actually placed above the NativeControlHost always disappear into its background. WinForms methods like 'SendToBack()' have no effect.

My acute struggle is that I place the NativeControlHost object in a ScrollView, but the NativeControl overlays the ScrollBars so they are not visible. See screenshot

![Screenshot](https://github.com/AvaloniaUI/Avalonia/assets/12414197/d1b6ab2c-5176-4694-b323-124d0ce03279)

XAML code of the window looks like this:
`
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
        x:Class="netgoconnect.GUI.Views.RdConnectionWindowNormal"
        Title="netgo connect - Remotedesktopverbindung"
        xmlns:viewModels="clr-namespace:netgoconnect.GUI.ViewModels"
        Background="{x:Null}"
        Icon="/Assets/img/icon_netgo-normal.png"
        SystemDecorations="Full"
        WindowState="Normal"
        IsVisible="False"
        CanResize="True"
        Width="800"
        Height="400">
    
    <Design.DataContext>
        <viewModels:RdConnectionWindowViewModel />
    </Design.DataContext>
    
    <Grid>
        <ScrollViewer HorizontalScrollBarVisibility="Visible" VerticalScrollBarVisibility="Visible"
                      Viewport=""
                      AllowAutoHide="False">
            
            <ContentControl Padding="0,200,0,200">
                
                <ContentControl Content="{Binding Content}"  // content contains the NativeControlHost control
                Width="{Binding VirtualScreenWidth}"
                Height="{Binding VirtualScreenHeight}" />
            </ContentControl>
        </ScrollViewer>
    </Grid>
</Window>
`

Do you have any ideas or a hack how I can make my Avalonia components more dominant?

**Comment by timunie** [comment, marked as answer]
It's the same as in WPF... search for "airspace issue"
reference: https://github.com/dotnet/wpf/issues/152

**Reply by StayMeaty** [reply]
Very good hint. Thank you!
